frontend/apps/reader/modules/readertypography.lua

```diff
@@ -718,6 +763,15 @@ function ReaderTypography:onReadSettings(config)
     self.ui.document:setHyphLeftHyphenMin(G_reader_settings:readSetting("hyph_left_hyphen_min") or 0)
     self.ui.document:setHyphRightHyphenMin(G_reader_settings:readSetting("hyph_right_hyphen_min") or 0)
 
+    -- Default to disable hanging/floating punctuation
+    -- (Stored as 0/1 in docsetting for historical reasons, but as true/false
+    -- in global settings.)
```

Comment:
That's the "reason" behind the funky setting.

It's a really old quirk, so much so that I don't remember exactly why/when that quirk was implemented ^^.